build.sh

```diff
@@ -472,6 +493,95 @@ function doDockerBuild {
              "${DOCKER_RUN_ARGS[@]}"
 }
 
+# Performs the build using the build server, zipping up the input, sending it to the server, and
+# unzipping the output.
+# Output: the $HTML_OUTPUT directory will contain the built files
+function doServerBuild {
+  clearDir "$HTML_TEMP"
+
+  local input_zip="build-server-input.zip"
+  local build_server_output="build-server-output"
+  local build_server_headers="build-server-headers.txt"
+
+  # Keep include list in sync with `processSource`
+  #
+  # We use an allowlist (--include) instead of a blocklist (--exclude) to avoid accidentally
+  # sending files that the user might not anticipate sending to a remote server, e.g. their
+  # private-notes-on-current-pull-request.txt.
+  #
+  # The contents of fonts/, images/, and dev/ are not round-tripped to the server, but instead
+  # copied below in this function. (We still send the directories to avoid the build script on the
+  # server getting confused about their absence.) demos/ needs to be sent in full for inlining.
+  local zip_args=(
+    --recurse-paths "$HTML_TEMP/$input_zip" . \
+    --include ./source ./404.html ./link-fixup.js ./html-dfn.js ./styles.css \
+              ./fonts/ ./images/ ./dev/ ./demos/\*
+  )
+  $QUIET && zip_args+=( --quiet )
+  (cd "$HTML_SOURCE" && zip "${zip_args[@]}")
+
+  local query_params=()
+  $QUIET && query_params+=( quiet )
+  $VERBOSE && query_params+=( verbose )
+  $DO_UPDATE || query_params+=( no-update )
+  $DO_LINT || query_params+=( no-lint )
+  $DO_HIGHLIGHT || query_params+=( no-highlight )
+  $SINGLE_PAGE_ONLY && query_params+=( single-page )
+
+  $QUIET || echo
+  $QUIET || echo "Sending files to the build server..."
+
+  local query_string
+  query_string="$(joinBy "\&" "${query_params[@]}")"
```

Comment:
Similar to the one I got with the PR for removing the perl scripts, I’m getting an error in my environment:
```
./build.sh: line 535: query_params[@]: unbound variable
```
So I guess we should either do this:
```suggestion
  query_string="$(joinBy "\&" "${query_params[@]-''}")"
```
…or alternatively something else to ensure that that `query_params` array isn’t empty